scipy/stats/_qmc.py

```diff
@@ -1810,25 +1810,33 @@ def __init__(
 
         super().__init__(d=1, seed=seed)
 
-    def random(self, n: IntNumber = 1) -> np.ndarray:
+    def random(  # type: ignore[override]
+        self, k: IntNumber, n: IntNumber = 1
+    ) -> np.ndarray:
         """Draw `n` QMC samples from the multinomial distribution.
 
         Parameters
         ----------
+        k : int
```

Comment:
The code is fine, but I would have expected this to be a parameter of the class initializer, as it is a shape parameter of the distribution. (Then `n` would be an OK name, unless `n` is never to be used as the name of a shape parameter for any distribution, since it is the name of `random`'s parameter. That would be unfortunate.) I think as we add QMC distributions, it will be important to enforce this distinction to keep the API regular, even when it is not computationally expensive to allow shape parameters in `random`.
If I could change just one thing about `rv_generic`, I think that would be it: embracing OOP and separating the creation of a distribution with shape parameters from invocation of the methods.

Reply:
I totally agree with putting this in the init. I would also prefer to not have this frozen dist concept and just have a pure class based API for dists.

Ok then for this one, either we make a breaking changes by directly moving the param to the init (it's a bug right now as it does not really do what should be expected). Or I move the param, and also allow it in random for backward compatibility. This would raise a warning.

What do you think?

Reply:
>  I would also prefer to not have this frozen dist concept

Funny that you should mention that: https://github.com/tirthasheshpatel/scipy/pull/14#issuecomment-1100914807

> Ok then for this one, either we make a breaking changes by directly moving the param to the init 

I think it's going to be backward compatible either way because the first argument, and the only one allowed as a positional argument, should be the number of samples, not the number of trials.

I'd just move it for now.

Reply:
> > I would also prefer to not have this frozen dist concept
> 
> Funny that you should mention that: [tirthasheshpatel#14 (comment)](https://github.com/tirthasheshpatel/scipy/pull/14#issuecomment-1100914807)

haha +1000 for this. Do you plan to bring this up?

> > Ok then for this one, either we make a breaking changes by directly moving the param to the init
> 
> I think it's going to be backward compatible either way because the first argument, and the only one allowed as a positional argument, should be the number of samples, not the number of trials.
> 
> I'd just move it for now.

It is yes. The current code is just a shape incompatibility though. So less hard than breaking the initialisation of the class. But since it's wrong and this class pretty new, better break things now than having a complicated update path for users.

I will do the changes.

Reply:
I did the update. I chose to use `n_trials` to make the name explicit and not redundant with `n` from random. In the multinomial and binomial distribution classes `n` and `p` are used. But not very clear for non stats people (well the description of the class are very not user friendly too).

Reply:
> I'd just move it for now.

If you don't feel strongly about it, I think we should double down on the decision to have shape parameters specified only in the initializer. I was just offering the option of accepting the number of trials as a keyword if you felt strongly that it belonged in the `random` method. Either way, I was saying that this PR should just move the parameter, not add it back to `random` as a keyword.

Reply:
I gave it more thoughts and I agree that any "shape" parameters should not be available elsewhere than during initialization 👍 This way, `random` is also more consistent with other classes as it just sample `n`.

Reply:
> Do you plan to bring this up?

Yes.

Reply:
> I gave it more thoughts and I agree that any "shape" parameters should not be available elsewhere than during initialization

@tupui so would you like to remove `n_trials` before this is merged?

Reply:
I already moved it in the init. It's not anymore in `random`. Or I misunderstood something?

Reply:
Oops, it's fine. I thought we would still call it `n` if it is in the initializer. I'll take a final look and merge today.